mmv1/third_party/terraform/resources/resource_container_node_pool.go.erb

```diff
@@ -378,9 +378,25 @@ func resourceContainerNodePoolCreate(d *schema.ResourceData, meta interface{}) e
 	if err != nil {
 		return err
 	}
-
+<% if version == "ga" -%>	
 	mutexKV.Lock(nodePoolInfo.lockKey())
 	defer mutexKV.Unlock(nodePoolInfo.lockKey())
+<% else -%>
+	clusterHash, err := getClusterHash(nodePoolInfo.parent(), nodePoolInfo.project, userAgent, config)
```

Comment:
What's the difference between using the relative name of the cluster and using the hash here? My understanding is that both will prove sufficient to identify the cluster's uniqueness, and that we make one less API call per mutation by not fetching the hash.

Reply:
Sorry for the slow reply here - basically, the {location, project, name} is unique for the lifespan of the cluster, but it can be deleted and a new, different cluster with the same {location, project, name} can be created. The most straightforward way to distinguish between these two distinct clusters is by using the cluster hash. 

However, for Terraform specifically, we may not actually care about this specific case. If that is the case, we can simply use something like `"projects/<project>/locations/<location>/clusters/<clusterName>/nodePools/<nodePoolName>"` and remove this API call.

Reply:
I think we can omit the call- it won't be possible for the node pool's parent cluster to change under the hood w/o the node pool changing as well. (As a whole, Terraform wouldn't recognise it if the underlying cluster changed out-of-band)

Reply:
Makes sense, I removed the API call to get the cluster hash and instead am using the `"projects/<project>/locations/<location>/clusters/<clusterName>/nodePools/<nodePoolName>"` format described above as the node pool lock key format.